### PR TITLE
Add documentation linking to sqlalchemy

### LINKED ISSUE #23086: It is not clear whether database connection need to be closed manually after calling pd.read_sql()
The document does not mention it.

### PR DESCRIPTION
**Pull request by cruzzoe**
- [x] closes #23086
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by alimcmaster1** [comment]
Hey - Can you submit updates on the same PR https://github.com/pandas-dev/pandas/pull/29363 in the future - means we don’t have to re-review..

**Comment by cruzzoe** [comment]
Sure. Sorry, didn't know what the correct process was.

File: pandas/core/generic.py [hunk 1]
```diff
@@ -2646,6 +2646,10 @@ def to_sql(
         con : sqlalchemy.engine.Engine or sqlite3.Connection
             Using SQLAlchemy makes it possible to use any DB supported by that
             library. Legacy support is provided for sqlite3.Connection objects.
+
+            Closing the connection is handled by the SQLAlchemy Engine. See `here \
```
Comment (WillAyd) [inline pandas/core/generic.py]: Is this continuation necessary? Don't think we have anywhere else?

**Comment by cruzzoe** [comment]
Yes. Without the continuation it would not render correctly

**Review by WillAyd — CHANGES_REQUESTED** [review]
Might be a personal thing but I find the wording here rather confusing. By saying `Closing the connection is handled by the SQLAlchemy Engine` I got the impression it was done automatically, but that is not the case. Any chance you can clarify or reword that to make it clear that pandas does nothing special here, i.e. the user is responsible for management of the connection?

There was a previous PR #27972 that was pretty close to getting merged if you'd like to take a look

**Comment by alimcmaster1** [comment]
I'm thinking we do something like this - similar to the other PR:

image

@WillAyd do you agree? I also got confused by current wording.

**Comment by WillAyd** [comment]
Yea that sounds like a great idea

**Comment by alimcmaster1** [comment]
LGTM (But I’m on mobile so can’t do a doc build to check how this renders).
CC @WillAyd

File: pandas/core/generic.py [hunk 2]
```diff
@@ -2645,7 +2645,11 @@ def to_sql(
             Name of SQL table.
         con : sqlalchemy.engine.Engine or sqlite3.Connection
             Using SQLAlchemy makes it possible to use any DB supported by that
-            library. Legacy support is provided for sqlite3.Connection objects.
+            library. Legacy support is provided for sqlite3.Connection objects. The user
+            is responsible for engine disposal and connection closure for the SQLAlchemy
+            connectable See `here \
+                <https://docs.sqlalchemy.org/en/13/core/connections.html>`_
```
Comment (alimcmaster1) [inline pandas/core/generic.py]: Not sure if the blank line is necessary

**Comment by WillAyd** [comment]
Content looks good. Can you build locally and just post a screenshot of how it renders to be safe?

**Comment by cruzzoe** [comment]
> Content looks good. Can you build locally and just post a screenshot of how it renders to be safe?

pd1
pd2

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @cruzzoe !

**Pull request merged**